codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -62,7 +62,7 @@ type t = {
 
 let shallowCopyPartition = (p: partition) => {
   ...p,
-  fetchedEventQueue: p.fetchedEventQueue->Array.copy,
+  fetchedEventQueue: p.fetchedEventQueue->Cow.copy,
```

Comment:
This is where the improvement happens. It won't copy the array until a mutation occurs